core/gather/driver/target-manager.js

```diff
@@ -115,6 +132,7 @@ class TargetManager extends ProtocolEventEmitter {
       const targetId = target.targetInfo.targetId;
       if (this._targetIdToTargets.has(targetId)) return;
 
+      if (isRootSession) this._mainFrameId = targetId;
```

Comment:
Is `targetId` guaranteed to be the frame id here? I think it would be more correct to get `_mainFrameId` via a call to `Page.getFrameTree` in the enable function.